### PR TITLE
rb: Fix side wide removal of an alias

### PR DESCRIPTION
**Pull request by vadmeste**
rb --force --dangerous should remove all buckets under an alias. This
PR fixes a recent regression.

### CODE REVIEW

**Comment by klauspost** [comment]
Can we add a regression test?

**Review by harshavardhana — APPROVED** [review]

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -35,6 +35,10 @@ jobs:
         env:
           GO111MODULE: on
         run: |
+          wget https://dl.min.io/server/minio/release/linux-amd64/minio && chmod +x minio
+          mkdir -p ~/.minio/certs/ && openssl req -x509 -newkey rsa:1024 -keyout ~/.minio/certs/private.key -out ~/.minio/certs/public.crt -days 7 -nodes -subj '/CN=localhost'
+          ./minio server /tmp/test-xl/{1...4}/ & sleep 10
+          export SERVER_ENDPOINT=localhost:9000 ACCESS_KEY=minioadmin SECRET_KEY=minioadmin ENABLE_HTTPS=1
```
Comment (harshavardhana) [inline .github/workflows/go.yml:41]: Shouldn't this export be in multiple lines?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -35,6 +35,10 @@ jobs:
         env:
           GO111MODULE: on
         run: |
+          wget https://dl.min.io/server/minio/release/linux-amd64/minio && chmod +x minio
+          mkdir -p ~/.minio/certs/ && openssl req -x509 -newkey rsa:1024 -keyout ~/.minio/certs/private.key -out ~/.minio/certs/public.crt -days 7 -nodes -subj '/CN=localhost'
```
Comment (harshavardhana) [inline .github/workflows/go.yml:39]: You can get the certs from minio-go repo if necessary for localhost keys @vadmeste Or commit the certs into mc repo.

File: functional-tests.sh [hunk 3]
```diff
@@ -456,7 +488,7 @@ function test_presigned_get_object()
     out=$("${MC_CMD[@]}" --json share download "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}")
     assert_success "$start_time" "${FUNCNAME[0]}" show_on_failure $? "unable to get presigned get object url"
     download_url=$(echo "$out" | jq -r .share)
-    curl -sSg --output "${object_name}.downloaded" -X GET "$download_url"
+    curl -ksSg --output "${object_name}.downloaded" -X GET "$download_url"
```
Comment (harshavardhana) [inline functional-tests.sh:491]: Let's just trust the system?

File: .github/workflows/go.yml (hunk 2 above)
Comment (harshavardhana) [reply .github/workflows/go.yml:39]: Or better use mkcert in batch mode to generate and trust the certs

File: functional-tests.sh (hunk 3 above)
Comment (harshavardhana) [reply functional-tests.sh:491]: You can avoid curl -k once certs trusted with mkcert

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by klauspost — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**